include/secp256k1.h

```diff
@@ -800,7 +817,7 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_pubkey_combine(
  *  implementations optimized for a specific tag can precompute the SHA256 state
  *  after hashing the tag hashes.
  *
- *  Returns 0 if the arguments are invalid and 1 otherwise.
+ *  Returns: 1 always.
```

Comment:
Wasn't sure why this has changed. Code below hasn't changed.

Reply:
The function really always returns 1:
https://github.com/bitcoin-core/secp256k1/blob/master/src/secp256k1.c#L750

I think this was just a copy-and-paste mistake when writing the API docs. Strictly speaking, it's not even an API change because there  simply are no invalid arguments for this function (except those catched by `ARG_CHECK` but then we'll die and the function won't return.)

Reply:
"We" are not guaranteed to die in an `ARG_CHECK` in case custom callbacks are used. Then the return value is unspecified in theory but 0 in practice. This is also how we describe the behavior of other functions (example `keypair_sec`, `keypair_pub`).

Reply:
I fixed also the other functions. See the commit message for detailed rationale.